lib/ide/ast/impl/src/lib.rs

```diff
@@ -588,158 +582,197 @@ pub enum MacroPatternMatchRaw<T> {
 // === AST ===
 // ===========
 
-// === HasSpan ===
 
-/// Things that can be asked about their span.
-pub trait HasSpan {
-    /// Length of the textual representation of This type in Unicode codepoints.
-    ///
-    /// Usually implemented together with `HasSpan`.For any `T:HasSpan+HasRepr`
-    /// for `t:T` the following must hold: `t.span() == t.repr().len()`.
-    fn span(&self) -> usize;
+// === Tokenizer ===
+
+/// An enum of valid Ast tokens.I
+#[derive(Debug)]
+pub enum Token<'a> { Off(usize), Chr(char), Str(&'a str), Ast(&'a Ast) }
+
+/// Things that can be turned into stream of tokens.
+pub trait HasTokens {
+    /// Feeds TokenBuilder with stream of tokens obtained from `self`.
+    fn feed(&self, consumer:&mut impl TokenConsumer);
```

Comment:
This doesnt make sense. How "HasTokens" is connected to "feed" ? How the fact that I have tokens is connected to the fact that I can be feed with tokens? Oh I see, the method name `feed` is wrong. Because it takes my tokens and pass it to the consumer - I would definitely work here on naming more.